File: codes/ruby/custom_error_and_error_handling.rb
print("\n", "# Custom Error and Error Handling in Ruby")

def give_me_respect(argument)
    # Custom Error
    raise StandardError.new("Error: You should give me \"respect\"!") if (argument != "respect")
    return "Thank you for giving me \"respect\"!"
end

# Error Handling
begin
    response = give_me_respect("boo!")
    print("\n", response)
rescue StandardError => an_error
    print("\n", an_error.message)
end
print("\n", "I'm sorry!")

# Without Error Handling
response = give_me_respect("boo!")
# this will never executed
print("\n", response)
print("\n", "I'm sorry!")

File: codes/javascript/src/custom_error_and_error_handling.js
console.log("\n// Custom Error and Error Handling in JavaScript");

const giveMeRespect = (argument) => {
    // Custom Error
    if (argument !== "respect") throw new Error('Error: You should give me "respect"!');
    return 'Thank you for giving me "respect"!';
};

let response;

// Error Handling
try {
    response = giveMeRespect("boo!");
    console.log(response);
} catch (anyError) {
    console.log(anyError.message);
}
console.log("I'm sorry!");

// Without Error Handling
response = giveMeRespect("boo!");
// this will never executed
console.log(response);
console.log("I'm sorry!");
